src/ValueCaster.php

```diff
@@ -32,7 +32,7 @@ public function castValue($value, array $allowedTypes)
         return new $castTo($value);
     }
 
-    public function castCollection($values, array $allowedArrayTypes)
+    public function castCollection($values, array $allowedArrayTypes, string $collectionClass = null)
```

Comment:
I'm not sure I like that this method's signature is growing. Any alternatives?

Reply:
Refactored